File: replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        
        for(int i = 0; i < arr.length; i++) {

           
           int cur_max = 0;


            for(int j = i + 1; j < arr.length; j++) {
                if(arr[j] > cur_max) {
                    cur_max = arr[j];
                    arr[i] = cur_max;
                }
            }
        }

        arr[arr.length - 1] = -1;
        return arr;

    }
}

File: replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        
        if len(arr) <= 1:
            return [-1]

        for i in range(len(arr)):
            if i + 1 < len(arr):
                arr[i] = max(arr[i + 1:len(arr)])

        
        arr[len(arr) - 1] = -1
        return arr